### PR TITLE
Add costomized banner message UI

### LINKED ISSUE #18719: Custom Banner messages to call-out maintenance and downtime
**Is your feature request related to a problem? Please describe.**
We have a requirement to call out downtime to the harbor server to the customers prior to the actual downtime, this could be done through a custom banner that can be closed by the users after reading or always visible on all pages.

**Describe the solution you'd like**
I would like to have a custom banner that can be displayed to users to inform them about planned downtime for the harbor server. The banner should have the following features:

    Clear and noticeable design: The banner should be designed in a way that catches the attention of users, ensuring they notice the important information.
    Customizable content: The content of the banner should be easily customizable to include specific details about the upcoming downtime, such as the date, time, and duration.
    Dismissal option: Users should have the ability to close the banner after reading it, so that it doesn't obstruct their view of the website.
    Persistent visibility: Alternatively, there should be an option to keep the banner visible on all pages throughout the downtime period, ensuring users are constantly reminded of the upcoming downtime.

**Describe the main design/architecture of your solution**
The solution can be implemented using a combination of front-end and back-end technologies. Here is a high-level overview of the design/architecture:

    Front-end:
        A banner component that can be rendered on all pages of the website.
        The banner should be styled to be visually prominent, with a clear message about the planned downtime.
        The content of the banner should be easily updatable, allowing administrators to customize the information displayed.

    Back-end:
        A configuration management system to store the details of the planned downtime, such as the date, time, and duration.
        An API endpoint to retrieve the downtime information from the configuration management system and provide it to the front end.
        An administration interface to allow authorized users to update the downtime details.

**Describe the development plan you've considered**
Here is a suggested development plan to implement the solution:

    Requirement analysis: Gather detailed requirements for the banner, including design preferences, customization options, and dismissal behavior.
    Front-end development:
        Implement the banner component with appropriate styling.
        Integrate the banner into the existing website template or content management system.
        Implement the dismissal functionality for the banner.
    Back-end development:
        Set up a configuration management system to store the downtime details.
        Implement an API endpoint to retrieve the downtime information.
        Develop an administration interface for authorized users to update the downtime details.
    Integration and testing:
        Integrate the front-end and back-end components.
        Perform thorough testing to ensure the banner functions as intended and is displayed correctly on all pages.
    Deployment:
        Deploy the updated website with the downtime banner to the production environment.
        Monitor the performance and user feedback to ensure the solution meets the requirements.
    Maintenance and updates:
        Regularly review and update the downtime details as needed.
        Address any user feedback or issues related to the banner.

**Additional context**

### PR DESCRIPTION
**Pull request by AllForNothing**
1.Fixes #18719
2.Add Banner Message item to configuration
3.Add banner_message property to systeminfo API

image

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/controller/systeminfo/controller.go [hunk 1]
```diff
@@ -90,11 +91,18 @@ func (c *controller) GetInfo(ctx context.Context, opt Options) (*Data, error) {
 		logger.Errorf("Error occurred getting config: %v", err)
 		return nil, err
 	}
+	mgr := config.GetCfgManager(ctx)
+	err = mgr.Load(ctx)
+	if err != nil {
+		logger.Errorf("Error occurred loading context: %v", err)
```
Comment (stonezdj) [inline src/controller/systeminfo/controller.go:97]: error occurred when loading configure

File: api/v2.0/swagger.yaml [hunk 2]
```diff
@@ -9088,6 +9096,11 @@ definitions:
         description: Whether or not to skip update pull time for scanner
         x-omitempty: true
         x-isnullable: true
+      banner_message:
+        type: string
+        description: The banner message for the UI.It is the stringified result of the banner message object
```
Comment (stonezdj) [inline api/v2.0/swagger.yaml:9101]: It's better to provide a sample value for it.

Comment (AllForNothing) [reply api/v2.0/swagger.yaml:9102]: Updated

File: src/controller/systeminfo/controller.go (hunk 1 above)
Comment (AllForNothing) [reply src/controller/systeminfo/controller.go]: Updated

**Review by stonezdj — APPROVED** [review]

**Review by chlins — APPROVED** [review]
lgtm

**Pull request merged**